Bio/SeqUtils/__init__.py

```diff
@@ -11,31 +11,139 @@
 
 
 import re
+import warnings
 from math import pi, sin, cos
 
 from Bio.Seq import Seq, complement, complement_rna
 from Bio.Data import IUPACData
-
+from Bio import BiopythonDeprecationWarning
 
 ######################################
 # DNA
 ######################
 # {{{
 
+_ambiguous_gc_values = {
+    "A": 0,
+    "C": 1.0,
+    "T": 0,
+    "G": 1.0,
+    "S": 1.0,
+    "V": 2 / 3,
+    "B": 2 / 3,
+    "M": 0.5,
+    "R": 0.5,
+    "Y": 0.5,
+    "K": 0.5,
+    "X": 0.5,
+    "N": 0.5,
+    "H": 1 / 3,
+    "D": 1 / 3,
+    "W": 0,
+}
+
+
+def gc_content(seq, ambiguous="remove"):
+    """Calculate G+C percentage in seq (float between 0 and 1).
+
+    Copes with mixed case sequences. Ambiguous Nucleotides in this context are
+    those different from ATCGSW.
+
+    If ambiguous equals "remove" (default), will only count GCS towards the
+    percentage, but will also only count ACTGSW towards the length of the sequence.
+    Equivalent to gc_content(seq.replace('N','')) but replacing all nucleotides
+    that are ambiguous between GC or AT (VBMRYKXNHD).
```

Comment:
Maybe:

```
Equivalent to gc_content(seq.replace('N','')) but removing not just N but all
nucleotides that are ambiguous between GC or AT (VBMRYKXNHD).
```

Reply:
If ambiguous equals "remove" (default), will only count GCS, and will only include ACTGSW when calculating the sequence length.

Reply:
You may want to remind the reader here that S is G or C, and W is A or T.

Reply:
+1. That would leave this as the paragraph (not checked for best line wrapping):

```
If ambiguous equals "remove" (default), will only count GCS (where S means G or C),
and will only include ACTGSW (where W means A or T) when calculating the sequence
length. Equivalent to gc_content(seq.replace('N','')) but removing not just N but all
nucleotides that are ambiguous between GC or AT (VBMRYKXNHD).
```

Reply:
The last sentence
> Equivalent to gc_content(seq.replace('N','')) but replacing all nucleotides that are ambiguous between GC or AT (VBMRYKXNHD).
sounds a bit confusing. I would leave it out.

Reply:
I also flagged that last sentence. Did you like my suggestion better?

``Equivalent to gc_content(seq.replace('N','')) but removing not just N but all nucleotides that are ambiguous between GC or AT (VBMRYKXNHD).``

I want to explain the option name "remove" in the paragraph.

Reply:
The confusing part is the python snippet `gc_content(seq.replace('N',''))`. To me it reads like "Equivalent to this, but not really." How about "Equivalent to removing all characters in the set VBMRYKXNHD before calculating the GC content, as each of these ambiguous nucleotides can either be A,T or C,G."

Reply:
Yes that's better. The current sorting of VBMRYKXNHD is non-obvious, maybe alphabetic?

Reply:
Or be explicit:
```
    N Any nucleotide (A or C or G or T or U)
    R Purine (A or G)
    Y Pyrimidine (T or C)
    K Keto (G or T)
    M Amino (A or C)
    S Strong interaction (3 H bonds) (G or C)
    W Weak interaction (2 H bonds) (A or T)
    B Not A (C or G or T)
    D Not C (A or G or T)
    H Not G (A or C or T)
    V Not T or U (A or C or G)
```

Reply:
That seems too verbose, the IUPAC code is a well known standard after all.

Reply:
I added the explanation for each value as comments in the (renamed) `_gc_values` dict. Tried to incorporate the other suggestions